routing/blindedpath/blinded_path.go

```diff
@@ -109,6 +113,88 @@ type BuildBlindedPathCfg struct {
 	DefaultDummyHopPolicy *BlindedHopPolicy
 }
 
+// blindedHop holds the information about a hop we have selected for a blinded
+// path.
+type BlindedHop struct {
+	Vertex       route.Vertex
+	ChannelID    uint64
+	EdgeCapacity btcutil.Amount
+}
+
+// ValidateChainedChannels checks the partially specified blinded path.
+//
+// It verifies that:
+// - The ordering of the channels is valid.
+// - All the specified channels exist.
+// - All the involved nodes advertise route blinding.
+// - None of the involved nodes are included in the nodeOmissionSet.
+func ValidateChainedChannels(cfg *BuildBlindedPathCfg, target route.Vertex,
+	incomingChainedChannels []uint64,
+	nodeOmissionSet fn.Set[route.Vertex]) ([]BlindedHop, error) {
+
+	var (
+		node                    = target
+		otherNode, node1, node2 route.Vertex
+		partiallySpecifiedPath  = make([]BlindedHop, 0)
+	)
+
+	for _, chanID := range incomingChainedChannels {
+		channEdge, _, _, err := cfg.FetchChannelEdgesByID(chanID)
+
+		// If the channel can't be found, then an error is returned.
+		if err != nil {
+			return nil, err
+		}
+
+		node1 = channEdge.NodeKey1Bytes
+		node2 = channEdge.NodeKey2Bytes
+
+		switch {
+		case bytes.Equal(node[:], node1[:]):
+			otherNode = node2
+
+		case bytes.Equal(node[:], node2[:]):
+			otherNode = node1
+
+		// If the chain is not valid, then an error is returned.
+		default:
+			return nil, fmt.Errorf("specified path %v not found",
+				incomingChainedChannels)
+		}
+
+		// Return error if route blinding isn’t supported.
+		// Accepting blinded payments doesn’t mean the node supports
+		// routing them, so we don’t check the target node’s features.
```

Comment:
```suggestion
We do not check whether the target node advertises the routing blinding feature in the node announcement message because  the receiver might not want to signal forwarding of blinded payment, however the receiver can signal accepting blinded payment in the invoice feature bit, therefore we do not check the target node here.
```